x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -285,12 +335,37 @@ export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitializat
     mlLicense.fullLicenseAPIGuard(async (context, request, response) => {
       try {
         const { analyticsId } = request.params;
+
+        const { destinationIndex } = request.query;
+
+        // If user checks box to deleete the destinationIndex associcated with the job
+        if (destinationIndex) {
+          // Verify if user has privilege to delete the destination index
+
+          const userCanDeleteDestIndex = userCanDeleteIndex(context, destinationIndex);
```

Comment:
`/api/ml/ml_node_count` might be a good example of how this endpoint can be written.
If the delete index flag is true, you'll need to look up the destination index and if they don't have permission to delete, we should return a `response.forbidden();` immediately.

Reply:
Done in https://github.com/elastic/kibana/commit/d14fafab86d64ef3afce4bda0f3c415ea85e672e#diff-d4f061dec5ae8fcd9e871fa0ce0ed8bcR374